src/shared/conf-parser.c

```diff
@@ -602,7 +603,11 @@ int config_parse_many(
         if (r < 0)
                 return r;
 
-        return config_parse_many_files(conf_files, files, sections, lookup, table, flags, userdata, ret_stats_by_path);
+        r = config_parse_many_files(conf_files, files, sections, lookup, table, flags, userdata, ret_stats_by_path);
```

Comment:
We only set return values on function success, so we should add an error check and return early here if `config_parse_many_files()` failed.